### PR TITLE
Refs #34233 -- Used str.removeprefix()/removesuffix().

### PR DESCRIPTION
**Pull request by felixxm**
See https://peps.python.org/pep-0616/.

### CODE REVIEW

File: django/contrib/admin/views/main.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def get_ordering(self, request, queryset):
                         )
                     # reverse order if order_field has already "-" as prefix
                     elif order_field.startswith("-") and pfx == "-":
-                        ordering.append(order_field[1:])
+                        ordering.append(order_field.removeprefix("-"))
```
Comment (ngnpope) [inline django/contrib/admin/views/main.py:379]: Not sure whether it makes sense to rearrange this slightly?
```python
                    elif pfx == "-" and order_field.startswith(pfx):
                        ordering.append(order_field.removeprefix(pfx))
```

File: django/utils/datastructures.py [hunk 2]
```diff
@@ -277,7 +277,7 @@ def __getitem__(self, key):
         """
         use_func = key.startswith(self.prefix)
         if use_func:
```
Comment (ngnpope) [inline django/utils/datastructures.py:279]: We don't need this check anymore:
```suggestion
```

**Review by ngnpope — CHANGES_REQUESTED** [review]
I'd started looking at this too, but you've picked up on much more that I did. However, I found these too that you've missed:

```diff
diff --git a/django/contrib/sessions/backends/file.py b/django/contrib/sessions/backends/file.py
index b7a13d6d17..d3ba8b11dd 100644
--- a/django/contrib/sessions/backends/file.py
+++ b/django/contrib/sessions/backends/file.py
@@ -201,7 +201,7 @@ class SessionStore(SessionBase):
         for session_file in os.listdir(storage_path):
             if not session_file.startswith(file_prefix):
                 continue
-            session_key = session_file[len(file_prefix) :]
+            session_key = session_file.removeprefix(file_prefix)
             session = cls(session_key)
             # When an expired session is loaded, its file is removed, and a
             # new file is immediately created. Prevent this by disabling
diff --git a/django/contrib/staticfiles/finders.py b/django/contrib/staticfiles/finders.py
index 10fcb8ca4c..112a81d279 100644
--- a/django/contrib/staticfiles/finders.py
+++ b/django/contrib/staticfiles/finders.py
@@ -137,7 +137,7 @@ class FileSystemFinder(BaseFinder):
             prefix = "%s%s" % (prefix, os.sep)
             if not path.startswith(prefix):
                 return None
-            path = path[len(prefix) :]
+            path = path.removeprefix(prefix)
         path = safe_join(root, path)
         if os.path.exists(path):
             return path
diff --git a/django/contrib/staticfiles/handlers.py b/django/contrib/staticfiles/handlers.py
index b7d9b5c18d..cd1cde1fc6 100644
--- a/django/contrib/staticfiles/handlers.py
+++ b/django/contrib/staticfiles/handlers.py
@@ -42,7 +42,7 @@ class StaticFilesHandlerMixin:
         """
         Return the relative path to the media file on disk for the given URL.
         """
-        relative_url = url[len(self.base_url[2]) :]
+        relative_url = url.removeprefix(self.base_url[2])
         return url2pathname(relative_url)
 
     def serve(self, request):
diff --git a/django/contrib/staticfiles/storage.py b/django/contrib/staticfiles/storage.py
index 445cf6b954..d49326b6e4 100644
--- a/django/contrib/staticfiles/storage.py
+++ b/django/contrib/staticfiles/storage.py
@@ -231,7 +231,7 @@ class HashedFilesMixin:
             if url_path.startswith("/"):
                 # Otherwise the condition above would have returned prematurely.
                 assert url_path.startswith(settings.STATIC_URL)
-                target_name = url_path[len(settings.STATIC_URL) :]
+                target_name = url_path.removeprefix(settings.STATIC_URL)
             else:
                 # We're using the posixpath module to mix paths and URLs conveniently.
                 source_name = name if os.sep == "/" else name.replace(os.sep, "/")
diff --git a/django/core/handlers/wsgi.py b/django/core/handlers/wsgi.py
index c071b5ca8c..c2b7cc2b6f 100644
--- a/django/core/handlers/wsgi.py
+++ b/django/core/handlers/wsgi.py
@@ -187,7 +187,7 @@ def get_script_name(environ):
             # do the same with script_url before manipulating paths (#17133).
             script_url = _slashes_re.sub(b"/", script_url)
         path_info = get_bytes_from_wsgi(environ, "PATH_INFO", "")
-        script_name = script_url[: -len(path_info)] if path_info else script_url
+        script_name = script_url.removesuffix(path_info)
     else:
         script_name = get_bytes_from_wsgi(environ, "SCRIPT_NAME", "")
 
diff --git a/django/template/backends/django.py b/django/template/backends/django.py
index 218e5e0bc1..98abf7ddb5 100644
--- a/django/template/backends/django.py
+++ b/django/template/backends/django.py
@@ -104,7 +104,7 @@ def get_template_tag_modules():
 
         if hasattr(pkg, "__path__"):
             for name in get_package_libraries(pkg):
-                yield name[len(candidate) + 1 :], name
+                yield name.removeprefix(candidate).lstrip("."), name
 
 
 def get_installed_libraries():
diff --git a/django/test/testcases.py b/django/test/testcases.py
index 23148537f5..cd54dd2453 100644
--- a/django/test/testcases.py
+++ b/django/test/testcases.py
@@ -1551,7 +1551,7 @@ class FSFilesHandler(WSGIHandler):
 
     def file_path(self, url):
         """Return the relative path to the file on disk for the given URL."""
-        relative_url = url[len(self.base_url[2]) :]
+        relative_url = url.removeprefix(self.base_url[2])
         return url2pathname(relative_url)
 
     def get_response(self, request):
diff --git a/tests/forms_tests/field_tests/test_filepathfield.py b/tests/forms_tests/field_tests/test_filepathfield.py
index ca0f6f3a7a..092001b453 100644
--- a/tests/forms_tests/field_tests/test_filepathfield.py
+++ b/tests/forms_tests/field_tests/test_filepathfield.py
@@ -9,9 +9,7 @@ PATH = os.path.dirname(os.path.abspath(__file__))
 
 def fix_os_paths(x):
     if isinstance(x, str):
-        if x.startswith(PATH):
-            x = x[len(PATH) :]
-        return x.replace("\\", "/")
+        return x.removeprefix(PATH).replace("\\", "/")
     elif isinstance(x, tuple):
         return tuple(fix_os_paths(list(x)))
     elif isinstance(x, list):
```

**Comment by felixxm** [comment]
@ngnpope Thanks :+1: I pushed edits.

**Review by ngnpope — APPROVED** [review]
I wouldn't be at all surprised if we've missed some, but this is a good start.

Looks good 👍🏻

**Review by carltongibson — APPROVED** [review]
Yep. All these look good. Thanks 👍

**Comment by felixxm** [comment]
Thanks both :gift:

**Pull request merged**